pkg/controllers/staticroute/staticroute_controller.go

```diff
@@ -47,41 +47,20 @@ type StaticRouteReconciler struct {
 	StatusUpdater common.StatusUpdater
 }
 
-func (r *StaticRouteReconciler) listStaticRouteCRIDs() (sets.Set[string], error) {
-	staticRouteList := &v1alpha1.StaticRouteList{}
-	err := r.Client.List(context.Background(), staticRouteList)
-	if err != nil {
-		log.Error(err, "failed to list StaticRoute CRs")
-		return nil, err
-	}
-
-	CRStaticRouteSet := sets.New[string]()
-	for _, staticroute := range staticRouteList.Items {
-		CRStaticRouteSet.Insert(string(staticroute.UID))
-	}
-	return CRStaticRouteSet, nil
-}
-
 func (r *StaticRouteReconciler) deleteStaticRouteByName(ns, name string) error {
-	CRPolicySet, err := r.listStaticRouteCRIDs()
-	if err != nil {
-		return err
-	}
 	nsxStaticRoutes := r.Service.ListStaticRouteByName(ns, name)
 	for _, item := range nsxStaticRoutes {
-		uid := util.FindTag(item.Tags, commonservice.TagScopeStaticRouteCRUID)
-		if CRPolicySet.Has(uid) {
-			log.Info("skipping deletion, StaticRoute CR still exists in K8s", "staticrouteUID", uid, "nsxStatciRouteId", *item.Id)
-			continue
+		log.Info("Deleting StaticRoute", "Namespace", ns, "Name", name, "nsxStaticRouteId", *item.Id)
+		vpcInfo, err := commonservice.ParseVPCResourcePath(*item.Path)
+		if err != nil {
+			log.Error(err, "Failed to parse NSX VPC path for StaticRoute", "path", *item.Path)
+			return err
 		}
-
-		log.Info("deleting StaticRoute", "StaticRouteUID", uid, "nsxStaticRouteId", *item.Id)
-		path := strings.Split(*item.Path, "/")
-		if err := r.Service.DeleteStaticRouteByPath(path[2], path[4], path[6], *item.Id); err != nil {
-			log.Error(err, "failed to delete StaticRoute", "StaticRouteUID", uid, "nsxStaticRouteId", *item.Id)
+		if err := r.Service.DeleteStaticRouteByPath(vpcInfo.OrgID, vpcInfo.ProjectID, vpcInfo.VPCID, *item.Id); err != nil {
```

Comment:
I feel that we had better change https://github.com/vmware-tanzu/nsx-operator/blob/main/pkg/nsx/services/staticroute/staticroute.go#L107 DeleteStaticRouteByPath to pass a *model.StaticRoutes obj to do deletion like other components do. So, we don't need to handle VPC path logic here.

Reply:
Refactor the deletion function, please take another look, thanks!

Reply:
Thanks, Yanjun!